### PR TITLE
feature(tcomp): Add jms family tcomp and update tcomp BD archetype

### PR DESCRIPTION
**Pull request by StefanLMNG**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
Maven archetype for TCOMP DI

**What is the new behavior?**
Update Maven archetype for TCOMP BD
Add TCOMP JMS family (definition part)

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components-archetype/src/main/resources/archetype-resources/__artifactId__-definition/pom.xml [hunk 1]
```diff
@@ -2,159 +2,15 @@
...
-                            </instructions>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
-        </plugins>
-    </build>
+    <name>Components-${componentName}}-definition</name>
+    <artifactId>${artifactId}-definition</artifactId>
+    <packaging>pom</packaging>
```
Comment (pbailly) [inline components-archetype/src/main/resources/archetype-resources/__artifactId__-definition/pom.xml]: Change to "<packaging>jar</packaging>"

File: components-archetype/src/main/resources/archetype-resources/__artifactId__-runtime___runtimeVersion__/pom.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    <parent>
+        <groupId>org.talend.components</groupId>
+        <artifactId>components-${componentName}</artifactId>
+        <version>0.15.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>${componentName}-runtime_${runtimeVersion}</artifactId>
+    <packaging>pom</packaging>
```
Comment (pbailly) [inline components-archetype/src/main/resources/archetype-resources/__artifactId__-runtime___runtimeVersion__/pom.xml]: Change to "<packaging>jar</packaging>"

File: components-archetype/src/main/resources/archetype-resources/__artifactId__-runtime___runtimeVersion__/pom.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.talend.components</groupId>
+        <artifactId>components-${componentName}</artifactId>
+        <version>0.15.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
```
Comment (pbailly) [inline components-archetype/src/main/resources/archetype-resources/__artifactId__-runtime___runtimeVersion__/pom.xml]: Remove this line.

File: components-archetype/src/main/resources/archetype-resources/__artifactId__-definition/pom.xml [hunk 4]
```diff
@@ -2,159 +2,15 @@
...
-    <groupId>org.talend.components</groupId>
-    <artifactId>${artifactId}-parent</artifactId>
-    <version>0.14.0-SNAPSHOT</version>
-    <relativePath>../pom.xml</relativePath>
-   </parent>
+    <parent>
+        <groupId>org.talend.components</groupId>
+        <artifactId>${artifactId}-parent</artifactId>
+        <version>0.14.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
```
Comment (pbailly) [inline components-archetype/src/main/resources/archetype-resources/__artifactId__-definition/pom.xml]: Remove this line.

File: components-jms/jms-definition/src/test/java/org/talend/components/jms/JMSComponentFamilyDefinitionTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.talend.components.api.ComponentFamilyDefinition;
+import org.talend.components.api.ComponentInstaller;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.times;
+
+public class JmsComponentFamilyDefinitionTest{
```
Comment (pbailly) [inline components-jms/jms-definition/src/test/java/org/talend/components/jms/JMSComponentFamilyDefinitionTest.java]: The name of the file is incorrect "JMS" should not be capitalized

File: components-jms/jms-definition/src/main/java/org/talend/components/jms/output/JmsoutputProperties.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import org.talend.components.jms.JmsDatasetProperties;
+import org.talend.daikon.properties.presentation.Form;
+import org.talend.daikon.properties.property.Property;
+import org.talend.daikon.properties.property.PropertyFactory;
+
+
+import static org.talend.daikon.properties.property.PropertyFactory.newBoolean;
+import static org.talend.daikon.properties.property.PropertyFactory.newEnum;
+
+public class JmsOutputProperties extends ComponentPropertiesImpl {
```
Comment (pbailly) [inline components-jms/jms-definition/src/main/java/org/talend/components/jms/output/JmsoutputProperties.java]: The name of the file is invalid (the "O" of the word "Output" should be capitalized)

File: components-jms/jms-definition/src/test/java/org/talend/components/jms/JMSDatastoreDefinitionTest.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+// ============================================================================
+
+package org.talend.components.jms;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+import org.talend.components.api.component.runtime.RuntimeInfo;
+
+public class JmsDatastoreDefinitionTest {
```
Comment (pbailly) [inline components-jms/jms-definition/src/test/java/org/talend/components/jms/JMSDatastoreDefinitionTest.java]: The name of the file is incorrect "JMS" should not be capitalized

File: components-jms/jms-definition/src/test/java/org/talend/components/jms/JMSDatastorePropertiesTest.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        Widget property = advanced.getWidget("property");
+        assertThat(property, notNullValue());
+        Widget value = advanced.getWidget("value");
+        assertThat(value, notNullValue());
+    }
+
+    /**
+     * Checks {@link JmsDatastoreProperties#refreshLayout(Form)} doesn't hide userPassword and jqlWidget in initial state
+     */
+    @Test
```
Comment (pbailly) [inline components-jms/jms-definition/src/test/java/org/talend/components/jms/JMSDatastorePropertiesTest.java]: Fix this test.
It shouls look like:
```
@Test
    public void testRefreshLayout() {
        //TODO FIX ISSUE
        JmsDatastoreProperties properties = new JmsDatastoreProperties("test");
        properties.init();

        assertFalse(properties.getForm(Form.MAIN).getWidget("version").isHidden());
        assertFalse(properties.getForm(Form.MAIN).getWidget("contextProvider").isHidden());
        assertFalse(properties.getForm(Form.MAIN).getWidget("serverUrl").isHidden());
        assertTrue(properties.getForm(Form.MAIN).getWidget("userName").isHidden());
        assertTrue(properties.getForm(Form.MAIN).getWidget("userPassword").isHidden());

        properties.needUserIdentity.setValue(true);
        properties.refreshLayout(properties.getForm(Form.MAIN));
        assertFalse(properties.getForm(Form.MAIN).getWidget("version").isHidden());
        assertFalse(properties.getForm(Form.MAIN).getWidget("contextProvider").isHidden());
        assertFalse(properties.getForm(Form.MAIN).getWidget("serverUrl").isHidden());
        assertFalse(properties.getForm(Form.MAIN).getWidget("userName").isHidden());
        assertFalse(properties.getForm(Form.MAIN).getWidget("userPassword").isHidden());
    }
```

File: components-jms/jms-definition/src/main/java/org/talend/components/jms/JmsDatastoreProperties.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    public Property<String> https_settings = PropertyFactory.newString("https_settings");
+
+    public Property<String> property = PropertyFactory.newString("property","");
+
+    public Property<String> value = PropertyFactory.newString("value","");
+
+    @Override
+    public void setupLayout() {
+        super.setupLayout();
+        Form mainForm = new Form(this, Form.MAIN);
```
Comment (pbailly) [inline components-jms/jms-definition/src/main/java/org/talend/components/jms/JmsDatastoreProperties.java]: Add the userName and userPassword to the main Form definition

**Comment by ftom** [comment]
Your license header is also an old one (2015). Can you refresh to the 2016 one ?

File: components-jms/jms-definition/src/main/java/org/talend/components/jms/JmsDatastoreProperties.java [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+    public Property<Boolean> needUserIdentity = newBoolean("needUserIdentity", false);
+
+    // TODO check if it is not better to do "UserPasswordProperties" class like for cassandra
+    public Property<String> userName = PropertyFactory.newString("userName","");
+
+    public Property<String> userPassword = PropertyFactory.newString("userPassword","");
+
+    // Those advanced settings could be either in the datastore or in the dataset
+    public Property<Boolean> use_https = PropertyFactory.newBoolean("use_https",false);
```
Comment (ftom) [inline components-jms/jms-definition/src/main/java/org/talend/components/jms/JmsDatastoreProperties.java]: Can you align the attribute names with standard camel case naming as above ?

File: components-jms/jms-definition/src/main/java/org/talend/components/jms/output/JmsoutputProperties.java [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+    public enum JmsAdvancedDeleveryMode {
```
Comment (ftom) [inline components-jms/jms-definition/src/main/java/org/talend/components/jms/output/JmsoutputProperties.java]: typo : s/Delevery/Delivery/

**Pull request merged**